packages/shared/package.json

```diff
@@ -2,25 +2,58 @@
   "name": "@clerk/shared",
   "version": "0.24.4",
   "description": "Internal package utils used by the Clerk SDKs",
-  "types": "./dist/types/index.d.ts",
-  "main": "./dist/cjs/index.js",
-  "module": "./dist/esm/index.js",
   "sideEffects": false,
   "files": [
     "dist"
   ],
+  "main": "./dist/index.js",
+  "module": "./dist/index.mjs",
   "exports": {
     ".": {
-      "types": "./dist/types/index.d.ts",
-      "import": "./dist/esm/index.js",
-      "require": "./dist/cjs/index.js"
+      "import": {
```

Comment:
This is the core logic of this PR.

- `.` allows for named import from `@clerk/shared`
- `./*` allows for import from `@clerk/shared/<name>`
- `./react` allows for named import from `@clerk/shared/react`
- `./react/hooks` allows for named import from `@clerk/shared/react/hooks`

You won't be able to import certain things from the top-level `@clerk/shared` because they are not exported on purpose inside the `index` file. You can access them through the subpath though.

Reply:
> You won't be able to import certain things from the top-level @clerk/shared because they are not exported on purpose inside the index file

Are we sure about this? Since the shared package is not a private package and may be used by some customers implementing their own flows, wouldn't it be a breaking change?

Reply:
Yes, we should probably make it a major version. I can simplify it by putting it all into `react`, I tried that in the WIP stage and kept it for now

Reply:
It should be a major version to avoid accidentally having an older version of our packages using `^` in dependencies, installing the latest version, and break.